main/saturn-node.js

```diff
@@ -79,22 +78,17 @@ async function start () {
 
   /** @type {Promise<void>} */
   const readyPromise = new Promise(function startSaturnNodeChildProcess (resolve, reject) {
-    if (!childProcess) {
-      throw new Error('Unexpected error: child process is undefined after startup')
-    }
+    assert(childProcess, 'Unexpected error: child process is undefined after startup')
```

Comment:
I was using `if(!childProcess)` instead of `assert(childProcess)` to let silence TypeScript compiler errors. I didn't realise that `assert` type definitions are written in such a way that allows the compiler to understand that `childProcess` must be defined after the `assert` call returns.

Reply:
I just looked it up, good to know!

> Ultimately the goal of TypeScript is to type existing JavaScript constructs in the least disruptive way. For that reason, TypeScript 3.7 introduces a new concept called “assertion signatures” which model these assertion functions.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html